src/modules/75-cf/pages/onboarding/__test__/ValidatingYourFlag.test.tsx

```diff
@@ -0,0 +1,367 @@
+  },
+  features: [
+    {
+      archived: false,
+      createdAt: 1619219113160,
+      defaultOffVariation: 'false',
+      defaultOnVariation: 'true',
+      envProperties: {
+        defaultServe: { variation: 'true' },
+        environment: 'Mock_Environment',
+        modifiedAt: 1619219291549,
+        offVariation: 'false',
+        rules: [],
+        state: 'off',
+        variationMap: null,
+        version: 1
+      },
+      evaluation: '',
+      identifier: 'pastActiveFlag',
+      kind: 'boolean',
+      modifiedAt: 1619219113160,
+      name: 'pastActiveFlag',
+      owner: ['current auth user?'],
+      permanent: false,
+      prerequisites: [],
+      project: 'tnhucf1',
+      results: [],
+      status: { lastAccess: 1619219291549, status: 'active' },
+      tags: [],
+      variations: [
+        { identifier: 'true', name: 'True', value: 'true' },
+        { identifier: 'false', name: 'False', value: 'false' }
+      ]
+    }
+  ]
+}
+
+const renderComponent = (): RenderResult =>
+  render(
+    <TestWrapper
+      path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/onboarding/detail"
+      pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
+    >
+      <ValidateYourFlagView
+        flagInfo={{
+          project: 'dummy',
+          name: 'test-flag',
+          identifier: 'test_flag',
+          kind: 'boolean',
+          archived: false,
+          variations: [
+            { identifier: 'true', name: 'True', value: 'true' },
+            { identifier: 'false', name: 'False', value: 'false' }
+          ],
+          defaultOnVariation: 'true',
+          defaultOffVariation: 'false',
+          permanent: false
+        }}
+        language={SupportPlatforms[1]}
+        apiKey={{
+          name: 'xxx-xxx-xxx',
+          apiKey: 'xxx-xxx-xxx',
+          identifier: 'xxx-xxx-xxx',
+          type: 'server'
+        }}
+        environmentIdentifier="foo-123-bar"
+        testDone={false}
+        setTestDone={jest.fn()}
+      />
+    </TestWrapper>
+  )
+
+describe('ValidatingYourFlag', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+  test('Should be able to toggle Flag ON and it loads', async () => {
+    renderComponent()
+
+    const flagToggle = screen.getByTestId('flagToggle')
+    await waitFor(() => {
+      expect(flagToggle).toBeInTheDocument()
+      expect(flagToggle).not.toBeChecked()
+    })
+
+    userEvent.click(flagToggle)
+
+    const fetchingContainer = screen.getByTestId('fetchingContainer')
+
+    await waitFor(() => {
+      expect(fetchingContainer).toBeVisible()
+      expect(flagToggle).toBeChecked()
+    })
+  })
+
+  test('Should return error state if flag is not active', async () => {
+    renderComponent()
+
+    const flagToggle = screen.getByTestId('flagToggle')
+    await waitFor(() => {
+      expect(flagToggle).toBeInTheDocument()
+      expect(flagToggle).not.toBeChecked()
+    })
```

Comment:
No need for the `waitFor`
```suggestion
    const flagToggle = screen.getByTestId('flagToggle')
    expect(flagToggle).toBeInTheDocument()
    expect(flagToggle).not.toBeChecked()
```

Reply:
removed